src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -3170,18 +3170,23 @@ private static Expression CompareWithZero(DynamicMetaObject target, Func<Express
             return comparer(target.Expression.Cast(target.LimitType), ExpressionCache.Constant(0).Cast(target.LimitType)).Cast(typeof(object));
         }
 
-        private DynamicMetaObject CompareLT(DynamicMetaObject target,
-                                            DynamicMetaObject arg,
-                                            DynamicMetaObject errorSuggestion)
+        private DynamicMetaObject CompareLT(
+            DynamicMetaObject target,
+            DynamicMetaObject arg,
+            DynamicMetaObject errorSuggestion)
         {
             var enumerable = PSEnumerableBinder.IsEnumerable(target);
-            if (enumerable == null && (target.Value == null || arg.Value == null))
-            {
-                Expression result =
-                      target.LimitType.IsNumeric() ? CompareWithZero(target, Expression.LessThan)
-                    : arg.LimitType.IsNumeric() ? CompareWithZero(arg, Expression.GreaterThanOrEqual)
-                    : arg.Value != null ? ExpressionCache.BoxedTrue
-                    : ExpressionCache.BoxedFalse;
+            if (enumerable == null
+                && (LanguagePrimitives.IsNullLike(target.Value)
+                    || LanguagePrimitives.IsNullLike(arg.Value)))
+            {
+                Expression result = target.LimitType.IsNumeric()
+                    ? CompareWithZero(target, Expression.LessThan)
+                    : arg.LimitType.IsNumeric()
+                        ? CompareWithZero(arg, Expression.GreaterThanOrEqual)
+                        : !LanguagePrimitives.IsNullLike(arg.Value)
```

Comment:
Can you please reverse this ternary expression? Instead of `!<....> ? a : b`, make it `<> ? b : a`. It's easier to read the code that way.

Reply:
Done!